### PR TITLE
beacon initial implementation

### PR DESCRIPTION
**Pull request by IvanHdzF**
needs the on damage mechanic
has a parameter for stacks as a workaround

### CODE REVIEW

File: internal/weapons/claymore/beacon/beacon.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+type Weapon struct {
+	Index  int
+	stacks int
+}
+
+func (w *Weapon) SetIndex(idx int) { w.Index = idx }
+func (w *Weapon) Init() error      { return nil }
+
+func NewWeapon(c *core.Core, char *character.CharWrapper, p weapon.WeaponProfile) (weapon.Weapon, error) {
+	//After the character's Elemental Skill hits an opponent, their ATK will be increased by 20% for 8s.
```
Comment (imring) [inline internal/weapons/claymore/beacon/beacon.go:25]: where is atk% buff after using skill?

File: internal/weapons/claymore/beacon/beacon.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+			return mATK, true
+		},
+	})
+	char.AddStatMod(character.StatMod{
+		Base:         modifier.NewBase("beacon-hp", -1),
+		AffectedStat: attributes.HPP,
+		Amount: func() ([]float64, bool) {
+			if !c.Player.Shields.PlayerIsShielded() {
+				mHP[attributes.HPP] = 0.24 + float64(r)*.08
+			}
```
Comment (imring) [inline internal/weapons/claymore/beacon/beacon.go:62]: add attribute clearing `mHP[attributes.HPP] = 0` if the player has the shield

Comment (imring) [reply internal/weapons/claymore/beacon/beacon.go:62]: or use `return nil, false` if the player has the shield:
```go
	mHP := make([]float64, attributes.EndStatType)
	mHP[attributes.HPP] = 0.24 + float64(r)*.08
	char.AddStatMod(character.StatMod{
		Base:         modifier.NewBase("beacon-hp", -1),
		AffectedStat: attributes.HPP,
		Amount: func() ([]float64, bool) {
			if c.Player.Shields.PlayerIsShielded() {
				return nil, false
			}
			return mHP, true
		},
	})
```

**Comment by IvanHdzF** [comment]
Anything else?

File: internal/weapons/claymore/beacon/beacon.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+	stackDuration := 480 //8s * 60
+	const skillKey = "beacon-of-the-reed-sea-skill"
+	const damagedKey = "beacon-of-the-reed-sea-damaged"
+
+	mATK := make([]float64, attributes.EndStatType)
+	mHP := make([]float64, attributes.EndStatType)
+	mHP[attributes.HPP] = 0.24 + float64(r)*.08
+	char.AddStatMod(character.StatMod{
+		Base:         modifier.NewBase("beacon-of-the-reed-sea-atk", stackDuration),
```
Comment (imring) [inline internal/weapons/claymore/beacon/beacon.go:48]: `-1` instead of `stackDuration`

**Review by imring — APPROVED** [review]

**Pull request merged**